### PR TITLE
Take full advantage of capabilities of the new frzr version

### PR DESCRIPTION
**Pull request by NeroReflex**
This series of migrations scripts ensure proper intallation of an image that contains  frzr v1.0.0+

This also modifies chimeraos to make use of per-deployment overlays:
- the installed deployment is never unlocked (instead another subvolume mounted as an overlay will get modifications)
- the installed deployment is easy to reset
- modifications done in /etc and /var from the previous deployment will be carried on to the next one
- modifications done in /usr and /boot (after unlocking) will not be carried on

### CODE REVIEW

**Comment by NeroReflex** [comment]
This can only be merged when the frzr refactor will be merged

File: rootfs/usr/lib/frzr.d/install-0002-systemd-boot.migration [hunk 1]
```diff
@@ -1,6 +1,14 @@
+#! /bin/bash
+
 # Migrate to systemd-boot
+
+frzr_migration_version=0
+
 post_install() {
 	local MOUNT_PATH=$1
+	local SUBVOL=$2
```
Comment (pastaq) [inline rootfs/usr/lib/frzr.d/install-0002-systemd-boot.migration:9]: Formatting

File: rootfs/usr/lib/frzr.d/install-0003-root_part_gpt.migration [hunk 2]
```diff
@@ -0,0 +1,38 @@
+#! /bin/bash
+
+# Apply the correct GPT partition type to rootfs
+
+frzr_migration_version=1
+
+post_install() {
+    local MOUNT_PATH=$1
```
Comment (pastaq) [inline rootfs/usr/lib/frzr.d/install-0003-root_part_gpt.migration:8]: Formatting

File: rootfs/usr/lib/frzr.d/install-0004-overlays.migration [hunk 3]
```diff
@@ -0,0 +1,93 @@
+#! /bin/bash
+
+# Create the /etc and /var overlay
+
+frzr_migration_version=1
+
+post_install() {
+    local MOUNT_PATH=$1
+	local SUBVOL=$2
```
Comment (pastaq) [inline rootfs/usr/lib/frzr.d/install-0004-overlays.migration:9]: Formatting

File: rootfs/usr/lib/frzr.d/install-0005-kernel-overlays.migration [hunk 4]
```diff
@@ -0,0 +1,70 @@
+#! /bin/bash
+
+# Create the needed overlays to host a kernel compiled and installed by the "frzr kernel" tool
+
+frzr_migration_version=1
+
+post_install() {
```
Comment (pastaq) [inline rootfs/usr/lib/frzr.d/install-0005-kernel-overlays.migration:7]: Formatting

File: rootfs/usr/lib/frzr.d/install-0006-device-quirks.migration [hunk 5]
```diff
@@ -0,0 +1,44 @@
+#! /bin/bash
+
+# Create the /etc and /var overlay
+
+frzr_migration_version=1
+
+post_install() {
+    local MOUNT_PATH=$1
+	local SUBVOL=$2
```
Comment (pastaq) [inline rootfs/usr/lib/frzr.d/install-0006-device-quirks.migration:9]: Formatting

File: rootfs/usr/lib/frzr.d/install-0007-device-quirks-apply.migration [hunk 6]
```diff
@@ -0,0 +1,109 @@
+#! /bin/bash
+
+# Create the /etc and /var overlay
+
+frzr_migration_version=1
+
+post_install() {
+    local MOUNT_PATH=$1
```
Comment (pastaq) [inline rootfs/usr/lib/frzr.d/install-0007-device-quirks-apply.migration:8]: Formatting

File: rootfs/usr/lib/frzr.d/install-0009-initramfs-renamed.migration [hunk 7]
```diff
@@ -0,0 +1,71 @@
+#! /bin/bash
+
+# install the kernel and initramfs: this is done to avoid renaming initramfs files leading to kernel installs/uninstall not working properly
+
+frzr_migration_version=0
+
+post_install() {
+	local MOUNT_PATH=$1
+	local SUBVOL=$2
+    local NAME=$3
```
Comment (pastaq) [inline rootfs/usr/lib/frzr.d/install-0009-initramfs-renamed.migration:10]: Formatting

File: rootfs/usr/lib/frzr.d/install-0010-default-subvol.migration [hunk 8]
```diff
@@ -0,0 +1,24 @@
+#! /bin/bash
+
+# make the new deployment the default btrfs subvolume so that frzr kernel will work
+
+frzr_migration_version=0
+
+post_install() {
+	local MOUNT_PATH=$1
+	local SUBVOL=$2
+    local NAME=$3
```
Comment (pastaq) [inline rootfs/usr/lib/frzr.d/install-0010-default-subvol.migration:10]: Formatting

File: rootfs/usr/lib/frzr.d/uninstall-0001-overlays.removal [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+		;;
+	*)
+		return 1
+		;;
+	esac
+}
+
+pre_removal() {
+    local MOUNT_PATH=$1
+	local SUBVOL=$2
```
Comment (pastaq) [inline rootfs/usr/lib/frzr.d/uninstall-0001-overlays.removal:23]: Formatting

File: rootfs/usr/lib/frzr.d/install-0009-initramfs-renamed.migration (hunk 7 above)
Comment (pastaq) [inline rootfs/usr/lib/frzr.d/install-0009-initramfs-renamed.migration:10]: Formatting

File: rootfs/usr/lib/frzr.d/uninstall-0002-device-quirks.removal [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+		return 0
+		;;
+	*)
+		return 1
+		;;
+	esac
+}
+
+pre_removal() {
+    local MOUNT_PATH=$1
```
Comment (pastaq) [inline rootfs/usr/lib/frzr.d/uninstall-0002-device-quirks.removal:22]: Formatting

File: rootfs/usr/lib/frzr.d/install-0001-require_reboot.migration [hunk 11]
```diff
@@ -0,0 +1,11 @@
+#! /bin/bash
+
+# triggers notification for reboot in old BPM
```
Comment (pastaq) [inline rootfs/usr/lib/frzr.d/install-0001-require_reboot.migration:3]: Is this still required? Old BPM was removed from the client like a year ago

File: rootfs/usr/lib/frzr.d/install-0006-device-quirks.migration [hunk 12]
```diff
@@ -0,0 +1,44 @@
+#! /bin/bash
+
+# Create the /etc and /var overlay
```
Comment (pastaq) [inline rootfs/usr/lib/frzr.d/install-0006-device-quirks.migration:3]: Can we expound on the purpose of this migration here? It reads the same as 0004

File: rootfs/usr/lib/frzr.d/install-0001-require_reboot.migration (hunk 11 above)
Comment (NeroReflex) [reply rootfs/usr/lib/frzr.d/install-0001-require_reboot.migration:3]: > Is this still required? Old BPM was removed from the client like a year ago

I have no idea. I found it causing no harm and so I left it sitting there.

**Pull request merged**